lib/broadway_rabbitmq/producer.ex

```diff
@@ -402,26 +402,30 @@ defmodule BroadwayRabbitMQ.Producer do
             channel_ref: channel_ref
         }
 
-      {:error, {:auth_failure, 'Disconnected'}} ->
-        handle_backoff(state)
+      {:error, reason} ->
+        handle_connection_failure(state, reason)
+    end
+  end
 
-      {:error, {:socket_closed_unexpectedly, :"connection.start"}} ->
-        handle_backoff(state)
+  defp handle_connection_failure(state, reason) do
+    _ = Logger.error("Cannot connect to RabbitMQ broker: #{inspect(reason)}")
 
-      {:error, :econnrefused} ->
+    case reason do
+      {:auth_failure, 'Disconnected'} ->
         handle_backoff(state)
 
-      {:error, :unknown_host} ->
+      {:socket_closed_unexpectedly, :"connection.start"} ->
         handle_backoff(state)
 
-      {:error, :not_allowed} ->
+      reason when reason in [:econnrefused, :unknown_host, :not_allowed] ->
         handle_backoff(state)
+
+      reason ->
+        raise "unexpected error when connecting to RabbitMQ broker: #{inspect(reason)}"
```

Comment:
I tend to prefer logging the full reason and erroring with a short reason. that's because errors in supervision  trees are very nested and hard to read.

Reply:
Sure, no problem. Updated :)